orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/pipeline/ManualJudgmentStage.groovy

```diff
@@ -77,30 +77,58 @@ class ManualJudgmentStage implements StageDefinitionBuilder, RestartableStage, A
     @Override
     TaskResult execute(Stage stage) {
       def stageData = stage.mapTo(StageData)
+      def notificationState
+      def executionStatus
+      def outputs = [:]
+
       switch (stageData.state) {
         case StageData.State.CONTINUE:
-          return new DefaultTaskResult(ExecutionStatus.SUCCEEDED)
+          notificationState = NotificationState.manualJudgmentContinue
+          executionStatus = ExecutionStatus.SUCCEEDED
+          break
         case StageData.State.STOP:
-          def executionStatus = ExecutionStatus.TERMINAL
-          return new DefaultTaskResult(executionStatus)
+          notificationState = NotificationState.manualJudgmentStop
+          executionStatus = ExecutionStatus.TERMINAL
+          break
+        default:
+          notificationState = NotificationState.manualJudgment
+          executionStatus = ExecutionStatus.RUNNING
+          break
       }
 
-      def outputs = [:]
+      processNotifications(stage, stageData, outputs, notificationState)
+
+      return new DefaultTaskResult(executionStatus, outputs)
+    }
+
+    void processNotifications(Stage stage, def stageData, def outputs, NotificationState notificationState) {
```

Comment:
Would suggest typing the arguments and returning the outputs rather than modifying the `outputs` parameter as a side-effect.